src/main/kotlin/Menu.kt

```diff
@@ -0,0 +1,99 @@
+import java.util.InputMismatchException
+import java.util.Scanner
+
+open class Menu(val name: String, val scanner: Scanner) {
+
+    //дженерик-метод, добавляет новый объект в список
+    inline fun <reified T> addItem(list: MutableList<T>){
+        when (T::class.qualifiedName){
+            "Archive" -> {
```

Comment:
Здесь лучше сравнивать вот так:
```suggestion
        when (T::class) {
            Archive::class -> {
```

Reply:
Благодарю!!